File: Java/src/main/java/com/gildedrose/GildedRose.java
package com.gildedrose;

import com.gildedrose.items.CustomItemFactory;

class GildedRose {
    private final Item[] items;
    private final CustomItemFactory customItemFactory;

    public GildedRose(Item[] items) {
        this.items = items;
        this.customItemFactory = new CustomItemFactory();
    }

    public void updateQuality() {
        for (Item item : items) {
            customItemFactory.getCustomItem(item.name).updateQuality(item);
        }
    }

    public Item[] getItems() {
        return items;
    }
}

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt
package com.gildedrose

import com.gildedrose.items.CustomItemFactory

class GildedRose(var items: List<Item>) {

    companion object {
        val customItemFactory = CustomItemFactory()
    }

    fun updateQuality() {
        for (item in items) {
            customItemFactory.getCustomItem(item.name).updateQuality(item)
        }
    }
}

